### DISCUSSION TITLE
Comand for stern

### DISCUSSION THREAD

**Discussion by robinge330** [stern/stern#135] [Q&A] [answered]
Please could you guide the syntax to use
root@compute2:~# docker run ghcr.io/stern/stern:latest stern -n magma
invalid configuration: no configuration has been provided, try setting KUBERNETES_MASTER environment variable
root@compute2:~#

**Comment by rkmathi** [comment, marked as answer]
You have to pass your KUBECONFIG file to docker container and specify it.

If KUBECONFIG file is in `$HOME/.kube/config` , you can execute it like this.

```bash
$ docker run --rm \
  --mount type=bind,readonly,source=$HOME/.kube,target=/.kube \
  -e KUBECONFIG=/.kube/config \
  ghcr.io/stern/stern:latest -n default .
```

**Reply by robinge330** [reply]
root@compute1:~# docker run --rm --mount type=bind,readonly,source=$HOME/.kube,target=/.kube -e KUBECONFIG=/.kube/config ghcr.io/stern/stern:latest -n magma
Tail multiple pods and containers from Kubernetes

Usage:
  stern pod-query [flags]

Flags:
  -A, --all-namespaces             If present, tail across all namespaces. A specific namespace is ignored even if specified with --namespace.



This is the output of this
 yes $HOME/.kube/config does exists

**Reply by robinge330** [reply]
Please could you guide me what is missing

**Reply by robinge330** [reply]
On a ubuntu server what is the installation steps. the only one i could find is docker

**Reply by rkmathi** [reply]
You need to pass `pod-query` like this

```
docker run --rm \
  --mount type=bind,readonly,source=$HOME/.kube,target=/.kube \
  -e KUBECONFIG=/.kube/config \
  ghcr.io/stern/stern:latest -n magma .
```

( `.` means all pods)

**Reply by robinge330** [reply]
Thanks, this worked
